### PR TITLE
Feature/add aws launch template asg support

### PR DESCRIPTION
**Pull request by sheldonkwok**
I'm currently creating my AWS Auto Scaling Groups with Launch Templates instead of Launch Configurations. Launch Templates are extensible to launching single EC2 instances and Spot Fleets as well. This PR covers fetching the instance type with Launch Templates since it currently only checks for Launch Configurations.

I tested by fetching instance types on ASGs created with Launch Templates and Launch Configs.

### CODE REVIEW

**Comment by sheldonkwok** [comment]
Hmm it seems like the bot tagged the PR as XXL because of the vendor file changes.

**Comment by aleksandra-malinowska** [comment]
Thanks for sending this PR. Looks like this is a non-trivial change, so it should probably be reviewed by someone familiar with AWS cloud provider. cc @mumoshu @sethpollack @countspongebob

File: cluster-autoscaler/cloudprovider/aws/aws_manager.go [hunk 1]
```diff
@@ -50,6 +51,7 @@ const (
 // AwsManager is handles aws communication and data caching.
 type AwsManager struct {
 	service     autoScalingWrapper
+	ec2         ec2Wrapper
```
Comment (sethpollack) [inline cluster-autoscaler/cloudprovider/aws/aws_manager.go]: can we make these consistent? Maybe `ec2Service` and  `autoScalingService`?

File: cluster-autoscaler/cloudprovider/aws/aws_manager.go [hunk 2]
```diff
@@ -93,6 +95,7 @@ func createAWSManagerInternal(
 
 	manager := &AwsManager{
 		service:  *service,
+		ec2:      ec2Wrapper{ec2.New(session.New())},
```
Comment (sethpollack) [inline cluster-autoscaler/cloudprovider/aws/aws_manager.go]: Can we do the initialization of ec2Service and autoScalingService the same way?

**Comment by sheldonkwok** [comment]
@sethpollack addressed suggestions and added a test

File: cluster-autoscaler/cloudprovider/aws/aws_manager.go [hunk 3]
```diff
@@ -51,7 +51,7 @@ const (
 // AwsManager is handles aws communication and data caching.
 type AwsManager struct {
 	service     autoScalingWrapper
```
Comment (sethpollack) [inline cluster-autoscaler/cloudprovider/aws/aws_manager.go]: Can you rename `service` to `autoScalingService` as well?

**Comment by sethpollack** [comment]
Thanks!

**Comment by sheldonkwok** [comment]
@sethpollack updated in a rebase

**Comment by sethpollack** [comment]
LGTM Thanks!

**Comment by aleksandra-malinowska** [comment]
Thanks @sethpollack for review! 

@sheldonkwok it seems Godeps were updated manually and don't match the versions in kubernetes/kubernetes repo (e.g. aws-sdk-go is at 1.14.12 in k/k and 1.12.7-2-g35b3dd92 in this PR). Can you please follow [these steps](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-can-i-update-ca-dependencies-particularly-k8siokubernetes) to keep the dependencies consistent?

**Comment by sheldonkwok** [comment]
@aleksandra-malinowska ah I used `godep update github.com/aws/aws-sdk-go/...` since the current version of aws-sdk on autoscaler doesn't have Launch Templates. I will look at those steps.

**Comment by sheldonkwok** [comment]
@aleksandra-malinowska I just updated the godeps and modified the fix_gopath with the new dep. Was this done correctly?

File: cluster-autoscaler/fix_gopath.sh [hunk 4]
```diff
@@ -36,3 +36,12 @@ cp $GOPATH/src/k8s.io/autoscaler/cluster-autoscaler/_override/github.com/Azure/a
...
 git commit -a -m "Api override for AKS"
+
+echo Overridng AWS API
+
+aws_gopath_dir=$GOPATH/src/github.com/aws/aws-sdk-go/service
+mkdir -p $aws_gopath_dir
+cp -r $GOPATH/src/k8s.io/autoscaler/cluster-autoscaler/_override/github.com/aws/aws-sdk-go/service/* $aws_gopath_dir
+cd $aws_gopath_dir
+git add .
+git commit -a -m "Api override for AKS"
```
Comment (aleksandra-malinowska) [inline cluster-autoscaler/fix_gopath.sh]: s/AKS/AWS

**Comment by aleksandra-malinowska** [comment]
> the current version of aws-sdk on autoscaler doesn't have Launch Templates. I will look at those steps.

Does the current version in kubernetes/kubernetes, i.e. [1.14.12](https://github.com/kubernetes/kubernetes/blob/master/Godeps/Godeps.json#L292), support it? If yes, there's no need for override, just run the script and it should update it in autoscaler to this version.

**Comment by sheldonkwok** [comment]
It looks like it does. I was using the `kubernetes.sync` file. What command did you use to generate that?

**Comment by aleksandra-malinowska** [comment]
`kubernetes.sync` is supposed to be updated manually after Godeps update to the kubernetes/kubernetes commit that was used. This is for tracking at what revision we're synced. I'll update the FAQ to clarify this.

**Comment by sheldonkwok** [comment]
@aleksandra-malinowska  I updated the godeps and the kubernetes.sync. I had to make a change in main.go b/c of a kuberentes api change. Can you verify it? I mirrored the update in the kubernetes repo: https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-controller-manager/app/controllermanager.go#L207

**Comment by aleksandra-malinowska** [comment]
Looks good, thanks!

/lgtm

**Pull request merged**